src/rootPages/Designer/ui_choose_list.js

```diff
@@ -351,9 +351,20 @@ export default function (AB) {
             }
          );
 
+         this._handler_create = (def) => {
+            if (def?.type != "application") return;
+
+            if (this.AB.applications((app) => app.id == def.id).length < 1) {
+               // Add a new Application to AB._allApplications
+               this.AB.definitionSync("created", def.id, def);
+            }
+
+            this.loadData();
+         };
+
          this._handler_reload = (def) => {
```

Comment:
do we still need this function?

Reply:
Yes, I should rename that function to clarify